tests/test_media_handlers.py

```diff
@@ -205,10 +205,11 @@ async def on_post(self, req, resp):
 
     app.add_route('/', ResourceAsync() if asgi else Resource())
 
-    result = testing.simulate_get(app, '/')
+    # NOTE(caselit): force serialization in xml, since error.to_json uses the faulty handler
+    result = testing.simulate_get(app, '/', headers={'Accept': 'text/xml'})
```

Comment:
I'm not sure if error.to_json should better guard for this case.

Reply:
This was already pointed out by @vytas7 here https://github.com/falconry/falcon/pull/1768#issuecomment-683310866